### PR TITLE
Homework: Fibonacci

### PR DESCRIPTION
**Pull request by michalsoloducha**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: fibonacci
🏅 Granted 6 XP for @michalsoloducha

**Comment by ziobron** [comment]
⏱ 3 XP extra for @michalsoloducha for on-time delivery

File: homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,27 @@
 #pragma once
 
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
+    if (sequence == 1) return 1;
+    if (sequence == 2) return 1;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:5]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+        return result;
+    }
     return 0;
 }
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
+    // edge condition
+    if (sequence == 0) return 0;
+    if (sequence == 1) return 1;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:23]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
+    if (sequence == 2) return 1;
+    return fibonacci_recursive(sequence - 1) + fibonacci_recursive(sequence - 2);
     return 0;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:26]: dead code

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**